### PR TITLE
Add Environment Labels

### LINKED ISSUE #1438: Add labels to Environments
All environments should allow for the setting of multiple "labels" (similar to Task tags) that are serialized as metadata.

### PR DESCRIPTION
**Pull request by cicdw**
**Thanks for contributing to Prefect!**

Please describe your work and make sure your PR:

- [x] adds new tests (if appropriate)
- [x] updates `CHANGELOG.md` (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

Note that your PR will not be reviewed unless all three boxes are checked.

## What does this PR change?
This PR introduces a `labels` kwarg to all environments for specifying environment labels.  These labels are also serialized and communicated to Cloud. 

## Why is this PR important?
Closes #1438 - this PR is the first step in enabling first-class support for multi-environment agents.  Currently listed as an "Enhancement" in the Changelog, the actual corresponding Agent change will elevate this to a "Feature".

### CODE REVIEW

File: src/prefect/serialization/environment.py [hunk 1]
```diff
@@ -13,17 +13,22 @@ class BaseEnvironmentSchema(ObjectSchema):
     class Meta:
         object_class = Environment
 
+    labels = fields.List(fields.String())
```
Comment (jlowin) [inline src/prefect/serialization/environment.py]: If desired, these could be deserialized as a true `set` with a minimal custom field that wraps the `List` field:

```python
class Set(fields.List):
    def _deserialize(self, *args, **kwargs):
        return set(super()._deserialize(*args, **kwargs))

labels = Set(fields.String())
```

Comment (cicdw) [reply src/prefect/serialization/environment.py]: Based on my deserialization tests, this appears to pass through the `__init__` method and is converted to a set at deserialization time anyway.  If you think the custom field approach is more robust though, I'm happy to make the change

Comment (jlowin) [reply src/prefect/serialization/environment.py]: Oh I forgot about that -- totally fine then. Honestly I was looking at these tests when this occurred to me - https://github.com/PrefectHQ/prefect/pull/1442/files#diff-874215115fb45a0a8c40afed1ef73108R128 - I'm guessing the `set()` call is not needed then?

Comment (jlowin) [reply src/prefect/serialization/environment.py]: Oops never mind these are serialization tests!

Comment (jlowin) [reply src/prefect/serialization/environment.py]: Good to go!

**Pull request merged**